eos-paygd/main.c

```diff
@@ -24,14 +24,62 @@
 #include <libgsystemservice/service.h>
 #include <libeos-payg/service.h>
 #include <signal.h>
+#include <linux/reboot.h>
+#include <sys/reboot.h>
 
+/* Force a reboot in situations where we are not able to enforce PAYG */
+static gboolean
+sync_and_reboot (gpointer user_data)
+{
+    g_warning ("Forcing reboot now!");
```

Comment:
This is a nasty situation, and if it happens once it'll probably happen in a loop... however I think @treitter explicitly wants PAYG related warnings to be somewhat obscure so they don't leak any useful information if someone's triggering them on purpose.

We're also hindered by the fact that we have no idea why dbus or rootfs didn't come up, so this message can't really be incredibly clear.

That said, I wouldn't mind a little more verbosity, and losing the ! at the end.  Maybe @treitter can suggest better text, as he's been leading the charge on seemingly friendly obfuscation...

Reply:
Yes, as Derek raised, I think we should shut down here and in similar situations unless there's a high likelihood it could succeed upon a subsequent reboot. We really don't want to waste the power it takes to pointlessly reboot, especially for users on solar home systems where energy comes at a premium.

As far as the message content, my key concerns are that it's unique (so we can unambiguously find its origin in our code) and that we don't give too many clues (eg, don't prepend "eos-paygd" or "payg" to each message).

How about the convention of prepending `###] ` where `###` is a 3-digit number that hasn't already been used in the given git repo?

Reply:
I don't see how obscuring this makes much sense because (a) right now at least this repository is public and (b) you can see in the journal that it's coming from eos-paygd.

I'll change it to shutdown rather than reboot.

Reply:
In the spirit of the `xsss:` thing we did in https://phabricator.endlessm.com/T27174 I prefixed the message with four letters (`bcsn:`)

Reply:
In the spirit of the `xsss:` thing we did in https://phabricator.endlessm.com/T27174 I prefixed the message with four letters (`bcsn:`)